scripts/zkai_native_attention_mlp_adapter_compression_ablation_gate.py

```
@@ -0,0 +1,763 @@
+            raise AdapterCompressionAblationError(f"adapter route {key} mismatch")
+
+    accounting = _dict(sources["accounting"], "current accounting")
+    rows = _list(accounting.get("rows"), "current accounting rows")
+    if len(rows) != 1:
+        raise AdapterCompressionAblationError("current accounting row count drift")
+    row = _dict(rows[0], "current accounting row")
+    local = _dict(row.get("local_binary_accounting"), "current local accounting")
+    grouped = _dict(local.get("grouped_reconstruction"), "current grouped accounting")
+    current = VARIANTS[0]
+    if row.get("proof_json_size_bytes") != current["proof_json_bytes"]:
+        raise AdapterCompressionAblationError("current accounting proof JSON drift")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)